bundles/org.openhab.core.io.websocket.audio/src/main/java/org/openhab/core/io/websocket/audio/internal/PCMWebSocketAdapter.java

```diff
@@ -0,0 +1,148 @@
+    private final Set<PCMWebSocketConnection> webSocketConnections = Collections.synchronizedSet(new HashSet<>());
+
+    @Activate
+    public PCMWebSocketAdapter(BundleContext bundleContext, final @Reference AudioManager audioManager,
+            final @Reference AudioDialogProvider audioDialogProvider) {
+        this.bundleContext = bundleContext;
+        this.audioManager = audioManager;
+        this.audioDialogProvider = audioDialogProvider;
+        this.pingTask = executor.scheduleWithFixedDelay(this::pingHandlers, 10, 5, TimeUnit.SECONDS);
+    }
+
+    protected void onSpeakerConnected(PCMWebSocketConnection speaker) throws IllegalStateException {
+        synchronized (webSocketConnections) {
+            if (getSpeakerConnection(speaker.getId()) != null) {
+                throw new IllegalStateException("Another speaker with the same id is already connected");
+            }
+            webSocketConnections.add(speaker);
+            logger.debug("connected speakers {}", webSocketConnections.size());
+        }
+    }
+
+    protected void onClientDisconnected(PCMWebSocketConnection connection) {
+        logger.debug("speaker disconnected '{}'", connection.getId());
+        synchronized (webSocketConnections) {
+            webSocketConnections.remove(connection);
+            logger.debug("connected speakers {}", webSocketConnections.size());
+        }
+    }
+
+    public @Nullable PCMWebSocketConnection getSpeakerConnection(String id) {
+        synchronized (webSocketConnections) {
+            return webSocketConnections.stream()
+                    .filter(speakerConnection -> speakerConnection.getId().equalsIgnoreCase(id)).findAny().orElse(null);
+        }
+    }
+
+    @Override
+    public String getId() {
+        return ADAPTER_ID;
+    }
+
+    @Override
+    public Object createWebSocket(ServletUpgradeRequest servletUpgradeRequest,
+            ServletUpgradeResponse servletUpgradeResponse) {
+        logger.debug("creating connection");
+        return new PCMWebSocketConnection(this, executor);
+    }
+
+    @Deactivate
+    @SuppressWarnings("unused")
+    public synchronized void deactivate() {
+        logger.debug("stopping connection check");
+        pingTask.cancel(true);
+        disconnectAll();
+    }
+
+    private void pingHandlers() {
+        ArrayList<PCMWebSocketConnection> handlers = new ArrayList<>(webSocketConnections);
+        for (var handler : handlers) {
+            if (handler != null) {
+                boolean pinned = false;
+                var remote = handler.getRemote();
+                if (remote != null) {
+                    try {
+                        remote.sendPing(wrap("oh".getBytes(StandardCharsets.UTF_8)));
+                        pinned = true;
+                    } catch (IOException ignored) {
+                    }
+                }
+                if (!pinned) {
+                    logger.warn("ping failed, disconnecting speaker {}", handler.getId());
```

Comment:
This can happen whenever the speaker's websocket connection disappears, right?
So it is not a "bug situation" and hence we should only log it as debug, but not as warn.